internal/services/datafactory/data_factory_pipeline_resource.go

```diff
@@ -109,6 +110,69 @@ func resourceDataFactoryPipeline() *pluginsdk.Resource {
 			},
 		},
 	}
+
+	if features.FourPointOhBeta() {
```

Comment:
For future reference, these types of schema changes should use the inverted feature flag e.g. 
```
schema: map[string]*pluginsdk.Schema{
// We put the desired breaking schema changes in here
}

if !features.FourPointOhBeta {
// We patch over the breaking schema change with the old definition here
}
```

This makes cleaning up old/deprecated code post major release much easier since the only action is to delete the entire block wrapped in the `!features.FourPointOhBeta` flag instead of having to copy paste the new behaviour into the schema above.

However.. since we're planning to ship 4.0 this week there isn't any need for the flag at all, this is an exceptional week where you can go ahead and make the breaking change directly in the schema and it would be acceptable and fine to merge.

Reply:
@stephybun thanks for above.
Not relevant to the change required. Wanted to check what's the best practice to ensure the order is not checked when it comes to running `terraform plan`, e.g. I have made `parameters` a list of objects, when multiple `paramters` are declared, running plan will have diffs because of the order.
Looks like changing to TypeSet fix the problem, but not sure I should use TypeSet over TypeList.

Reply:
@hqhqhqhqhqhqhqhqhqhqhq it seems like the API doesn't preserve the order of the parameters, so we won't have any other choice but to use a Set here